src/operation/union/CoverageUnion.cpp

```diff
@@ -109,18 +111,23 @@ std::unique_ptr<Geometry> CoverageUnion::polygonize(const GeometryFactory* gf) {
 }
 
 std::unique_ptr<geom::Geometry> CoverageUnion::Union(const geom::Geometry* geom) {
-    CoverageUnion cu;
+    double area_in = geom->getArea();
+    const auto& factory = geom->getFactory();
 
-    cu.extractRings(geom);
-    cu.sortRings();
+    operation::cluster::GeometryIntersectsClusterFinder f;
+    operation::cluster::DisjointOperation op(f);
+    auto ret = op.processDisjointSubsets(*geom, [&factory](const geom::Geometry& g) {
```

Comment:
better to split this out into a separate PR, so it's more visible.   Would be nice to see metrics on performance increase in that PR.

Reply:
Sorry, my comment was not clear.  I mean move all changes to `CoverageUnion` to a separate PR, as a follow-up to this PR.  Likewise for changes to Geometry unary union.

Reply:
It's clear, I just haven't separated it yet.